libraries/plugins/elasticsearch/elasticsearch_plugin.cpp

```diff
@@ -0,0 +1,366 @@
+      visitor_data = ",\"fee_data\": " + fee_data + ", \"transfer_data\": " + transfer_data;
+   }
+
+   // block data
+   auto block = db.fetch_block_by_number(op_id(db).block_num);
+   std::string block_num = std::to_string(block->block_num());
+   std::string block_time = block->timestamp.to_iso_string();
+   std::string trx_id = "";
+   // removing by segfault at blok 261319,- static variant wich thing.
+   //if(!block->transactions.empty() && block->transactions[op_id(db).trx_in_block].id().data_size() > 0 && block->block_num() != 261319 ) {
+   //   trx_id = block->transactions[op_id(db).trx_in_block].id().str();
+   //}
+   std::string block_data = "{\"block_num\":" + block_num + ",\"block_time\":\"" + block_time +
+                             "\",\"trx_id\":\"" + trx_id + "\"}";
+
+   // check if we are in replay or in sync and change number of bulk documents accordingly
+   uint32_t limit_documents = 0;
+   if((fc::time_point::now() - block->timestamp) < fc::seconds(30))
+      limit_documents = _elasticsearch_bulk_sync;
+   else
+      limit_documents = _elasticsearch_bulk_replay;
+
+   //wlog((account_transaction));
+   //wlog((operation_history));
+   //wlog((fee_data));
+   //wlog((transfer_data));
+   //wlog((block_data));
+
+   if(bulk.size() < limit_documents) { // we have everything, creating bulk array
+
+      // put alltogether in 1 line of bulk
+      std::string alltogether = "{\"account_history\": " + account_transaction + ", \"operation_history\": " + operation_history +
+                         ",\"operation_type\": " + op_type + ", \"block_data\": " + block_data + visitor_data + "}";
+
+      // bulk header before each line, op_type = create to avoid dups, index id will be ath id(2.9.X).
+      std::string _id = fc::json::to_string(ath.id);
+      bulk.push_back("{ \"index\" : { \"_index\" : \"graphene\", \"_type\" : \"data\", \"op_type\" : \"create\", \"_id\" : "+_id+" } }");
+      bulk.push_back(alltogether);
+   }
+   std::string bulking = "";
+   if (curl && bulk.size() >= limit_documents) { // we are in bulk time, ready to add data to elasticsearech
+
+      bulking = boost::algorithm::join(bulk, "\n");
+      bulking = bulking + "\n";
+      bulk.clear();
+
+      //wlog((bulking));
+
+      struct curl_slist *headers = NULL;
+      curl_slist_append(headers, "Content-Type: application/json");
+      std::string url = _elasticsearch_node_url + "_bulk";
+      curl_easy_setopt(curl, CURLOPT_URL, url.c_str());
+      curl_easy_setopt(curl, CURLOPT_POST, true);
+      curl_easy_setopt(curl, CURLOPT_HTTPHEADER, headers);
+      curl_easy_setopt(curl, CURLOPT_POSTFIELDS, bulking.c_str());
+      curl_easy_setopt(curl, CURLOPT_WRITEFUNCTION, WriteCallback);
+      curl_easy_setopt(curl, CURLOPT_WRITEDATA, (void *)&readBuffer);
+      curl_easy_setopt(curl, CURLOPT_USERAGENT, "libcrp/0.1");
+      //curl_easy_setopt(curl, CURLOPT_VERBOSE, true);
+      curl_easy_perform(curl);
```

Comment:
Here you have to add error handling. In particular
* ES can reject the complete request with error 429 (too many requests)
* Even if the response is 200 OK the response body can contain failures of individual log entries, such a failure can contain code 429 as well
Generally, you should repeat the request for temporary failures, perhaps after some delay.
The question is how to handle permanent failures, i. e. error code 500 and up.